src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -198,15 +195,44 @@ private void AddPackageItemsToDictionary(ref Dictionary<string, PackageItem> pac
             }
         }
 
-        // A versions file is of the form https://github.com/dotnet/versions/blob/master/build-info/dotnet/corefx/master/Latest_Packages.txt
-        private Dictionary<string, PackageItem> GatherPackageInformationFromVersionsFile(string versionsFile, VersionComparer comparer = null)
+        private static async System.Threading.Tasks.Task<Stream> GetStream(Uri uri)
+        {
+            if(uri.Scheme == "file")
+            {
+                return new FileStream(uri.LocalPath, FileMode.Open, FileAccess.Read, FileShare.Read);
+            }
+            else
+            {
+                return await new HttpClient().GetStreamAsync(uri);
+            }
+        }
+
+        private static async System.Threading.Tasks.Task<string> GetFileAsync(Uri uri)
+        {
+            using (var stream = GetStream(uri).Result)
```

Comment:
`var stream = await GetStream(uri)`. In general you should avoid calling .Result when you are in an async method.